code/datums/diseases/black_goo.dm

```diff
@@ -1,100 +1,106 @@
+	var/stage_level = 0 // we start at zero when dead everything go twice as fast each stage take 6 min so it need 360 and if your dead it require 180 so 3 min
+	var/infection_rate = 1
+	stage_prob = 0//no randomness
+
 /datum/disease/black_goo/stage_act()
 	..()
 	if(!ishuman(affected_mob)) return
 	var/mob/living/carbon/human/H = affected_mob
 
-	if(age > 1.5*stage_minimum_age) stage_prob = 100 //if it takes too long we force a stage increase
-	else stage_prob = initial(stage_prob)
-	if(H.stat == DEAD) stage_minimum_age = 75 //the virus progress faster when the host is dead.
+	// check if your already a zombie or in the process of being transform into one...
+	if(H.species.name == SPECIES_ZOMBIE || zombie_transforming == TRUE)
+		return
+
+	// check if dead
+	if(H.stat == DEAD)
+		infection_rate = 4
+
+	// check if he isn't dead
+	if(H.stat != DEAD)
+		infection_rate = 1
+
+	// here we add the new infection rate to the stage level.
+	stage_level = stage_level + infection_rate
+
+	// we want to check if we have reach enough stage level to gain a stage 3 stage of 6 min if you get it once.
+	if(stage_level >= 360)
+		stage ++
+		stage_level = stage_level - 360
+
 	switch(stage)
+		//stage 1 : Viatiate "afflicted"
 		if(1)
 			if(H.stat == DEAD && stage_counter != stage)
-				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage One! Zombie transformation begins at Stage Four."))
+				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage One! Zombie transformation begins at Stage Three."))
 				stage_counter = stage
-			survive_mob_death = TRUE //changed because infection rate was REALLY horrible.
 			if(goo_message_cooldown < world.time )
-				if(prob(3))
-					to_chat(affected_mob, SPAN_DANGER("You feel really warm..."))
+				if(prob(5))
+					to_chat(affected_mob, SPAN_DANGER("You feel warm..."))
 					goo_message_cooldown = world.time + 100
+					stage_level = stage_level + 4
+				else if(prob(3))
+					to_chat(affected_mob, SPAN_DANGER("You can't trust them..."))
+					goo_message_cooldown = world.time + 100
+					stage_level = stage_level + 2
+
+		//stage 2 : Ague "Febrile"
 		if(2)
 			if(H.stat == DEAD && stage_counter != stage)
-				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage Two! Zombie transformation begins at Stage Four."))
-				stage_counter = stage
-			if(goo_message_cooldown < world.time)
-				if (prob(3)) to_chat(affected_mob, SPAN_DANGER("Your throat is really dry..."))
-				else if (prob(6)) to_chat(affected_mob, SPAN_DANGER("You feel really warm..."))
-				else if (prob(2)) H.vomit_on_floor()
-				goo_message_cooldown = world.time + 100
-		if(3)
-			if(H.stat == DEAD && stage_counter != stage)
-				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage Three! Zombie transformation begins at Stage Four, which will be soon."))
+				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage Two! Zombie transformation begins at Stage Three."))
```

Comment:
stage numbers aren't proper nouns.
Centering this seems 😕

Reply:
change Stage to stage and did the same for everything that had maj and shouldn't